head/firmware/step_motor.cpp

```diff
@@ -27,4 +27,13 @@ void start_motor_handler(
     handler_class.set_message_queue(queue, completed_queue);
 }
 
-void reset_motor_handler() { handler_class.reset(); }
\ No newline at end of file
+void reset_motor_handler() { handler_class.reset(); }
+
+MoveStatus get_move_status() {
+    auto buffered_move = handler_class.get_buffered_move();
```

Comment:
This is a data race with `buffered_move.velocity` - it can be altered during the copy at the return of `get_buffered_move`. Nothing is querying velocity, so it shouldn't affect anything, but it's kind of a bad habit. Let's at least note it down so nobody at some point tries to add velocity querying and gets weird results because of it.